### PR TITLE
Development

### PR DESCRIPTION
**Pull request by yohanology**
User stories done:

*USBAT create a service
*USBAT see list of services on homepage
*USBAT update his account settings
*USBAT see his public profile

and other changes:

*Seeds data created
*Design and layout improvement

### CODE REVIEW

**Comment by f3r** [comment]
Hey guys, would you be so kind to create one pull request per user story in the future?

**Comment by f3r** [comment]
Also, **before I merge this**

make sure you add to your `README.md` a new section that contain precise and user-friendly installation instructions (i.e.):

### Installation Instructions
- Fork the repo from: git@github.com:WDI-HK-8/XXXX.git
- `bundle install`
- `rake db:create`
- `rake db:migrate`
- `bower install`
- `rails server`

**Comment by yohanology** [comment]
yes sir we will! 没问题！

File: app/assets/javascripts/angular/controllers/auth/profile.js.erb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  $auth.validateUser()
+    .then(function(response) {
+      $scope.loggedIn = true;
+      $http.get(url + '/users/' + response.id).success(function(response) {
+        
+        $scope.profile = response;
+        
+        if (!response.image) {
+          $scope.profile.image = "https://lh3.googleusercontent.com/-hoI3rXz7oy0/AAAAAAAAAAI/AAAAAAAAAAA/UBf8PfFVAsQ/photo.jpg"
```
Comment (f3r) [inline app/assets/javascripts/angular/controllers/auth/profile.js.erb]: Add the image to your `assets/images` and then you can use [image_url](http://api.rubyonrails.org/classes/ActionView/Helpers/AssetUrlHelper.html] to generate the link:

``` js
$scope.profile.image = "<%= image_url("rails.png") %>"
```

imagine if googleusercontent.com goes down... you'll loose access to the image no?

**Comment by f3r** [comment]
`rake db:seed` is broken...

```
$ rake db:seed
rake aborted!
ActiveRecord::UnknownAttributeError: unknown attribute 'user_id' for Service.
/Users/f3r/GA/WDI/wdi8/project3/jimmythis/db/seeds.rb:27:in `block in <top (required)>'
/Users/f3r/GA/WDI/wdi8/project3/jimmythis/db/seeds.rb:23:in `each'
/Users/f3r/GA/WDI/wdi8/project3/jimmythis/db/seeds.rb:23:in `<top (required)>'
NoMethodError: undefined method `user_id=' for #<Service:0x007fc22e6f40d0>
/Users/f3r/GA/WDI/wdi8/project3/jimmythis/db/seeds.rb:27:in `block in <top (required)>'
/Users/f3r/GA/WDI/wdi8/project3/jimmythis/db/seeds.rb:23:in `each'
/Users/f3r/GA/WDI/wdi8/project3/jimmythis/db/seeds.rb:23:in `<top (required)>'
Tasks: TOP => db:seed
(See full trace by running task with --trace)
```

**Comment by yohanology** [comment]
README.md improved

**Comment by ctlong** [comment]
Seeding works for me.

**Comment by ctlong** [comment]
How is it now?

**Comment by ctlong** [comment]
How is it now?

File: app/assets/javascripts/angular/controllers/categories/show_category.js.erb [hunk 2]
```diff
@@ -0,0 +1,41 @@
+app.controller('CategoryShowCtrl', ['$scope','$http','$auth','$location','$routeParams', function($scope,$http,$auth,$location,$routeParams){
+  
+  var url = "<%= ENV['URL'] %>" || "http://localhost:3000";
+
+  $http.get(url + '/categories/' + $routeParams.id).success(function(response) {
+    console.log(response);
+
+    if(response.length > 0) {
+      $scope.services = response;
+      $scope.services.forEach(function(service) {
```
Comment (f3r) [inline app/assets/javascripts/angular/controllers/categories/show_category.js.erb]: this should definitely be done in the server side and add it to `categoryController#show` as an extra parameter that it returns

File: app/assets/javascripts/angular/controllers/services/manage_services.js.erb [hunk 3]
```diff
@@ -1,22 +1,24 @@
...
   $auth.validateUser()
     .then(function(resp) {
       $scope.loggedIn = true;
       $http.get(url + '/users/' + resp.id + '/services').success(function(response) {
-        if(response.services) {
-          $scope.services = response.services;
+        if(response.length > 0) {
+          $scope.services = response;
+          console.log($scope.services.ratings)
+          $scope.services.forEach(function(service) {
```
Comment (f3r) [inline app/assets/javascripts/angular/controllers/services/manage_services.js.erb]: same here, average of ratings would be better if it comes from the backend directly

File: app/assets/javascripts/angular/controllers/static_pages.js.erb [hunk 4]
```diff
@@ -1,21 +1,34 @@
...
-      console.log(response);
-    });
-  };
+  $http.get(url + '/services').success(function(response) {
+    console.log(response);
+
+    if(response) {
+      $scope.services = response;
+      $scope.services.forEach(function(service) {
+
```
Comment (f3r) [inline app/assets/javascripts/angular/controllers/static_pages.js.erb]: and one more time same same :)

File: app/assets/templates/auth/account.html [hunk 5]
```diff
@@ -1,39 +1,17 @@
...
-            <li><a href="/#/account">Account</a></li>
-            <li><a href="/#/services">Services</a></li>
-            <li><a href="/#/" ng-click="logout()">Log Out</a></li>
-          </ul>
-        </li>
-      </ul>
-    </div> 
-  </div>
-</nav>
+<navbar></navbar>
```
Comment (f3r) [inline app/assets/templates/auth/account.html]: naiseeee :+1:

File: app/assets/templates/services/create.html [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+          <option>Wan Chai</option>
+          <option>Wong Tai Sin</option>
+          <option>Yau Tsim Mong</option>
+        </select>
+      </div>
+
+      <div class="form-group">
+        <label>Category</label>
+        <select class="form-control" ng-model="category" required="required">
+          <option>Advertising</option>
```
Comment (f3r) [inline app/assets/templates/services/create.html:40]: in html, you should add `&` as `&amp;`

File: app/assets/templates/static_pages/index.html [hunk 7]
```diff
@@ -1,43 +1,32 @@
...
-            <li><a href="/#/account">Account</a></li>
-            <li><a href="/#/services">Services</a></li>
-            <li><a href="/#/" ng-click="logout()">Log Out</a></li>
-          </ul>
-        </li>
-      </ul>
-    </div> 
-  </div>
-</nav>
+<navbar></navbar>
```
Comment (f3r) [inline app/assets/templates/static_pages/index.html]: if all the views have a navbar... shouldn't you add it to https://github.com/WDI-HK-8/jimmythis/blob/development/app%2Fviews%2Flayouts%2Fapplication.html.erb?

**Comment by f3r** [comment]
Please merge after reviewing all comments, good job guys

**Pull request merged**